GRBR-463: Creation of a new Validation in SignaturePolicy validation …

fernandoraffani opened the pull request:
…chain, that makes the Formal Policy Validation if it is implemented by the Custom Validation (when the Policy is in ASN1 format and signature can be validated against it).

The implementation was made so that the Formal Validation is configurable in the xml and optional, with the default value being to skip validation, so it doesn't affect all the other policy that cannot have specific elements validated against the signature.

Commit:
GRBR-463: Creation of a new Validation in SignaturePolicy validation chain, that makes the Formal Policy Validation if it is implemented by the Custom Validation (when the Policy is in ASN1 format and signature can be validated against it).

The implementation was made so that the Formal Validation is configurable in the xml and optional, with the default value being to skip validation, so it doesn't affect all the other policy that cannot have specific elements validated against the signature.

Commit:
GRBR-463: Creation of a new Validation in SignaturePolicy validation chain, that makes the Formal Policy Validation if it is implemented by the Custom Validation (when the Policy is in ASN1 format and signature can be validated against it).

The implementation was made so that the Formal Validation is configurable in the xml and optional, with the default value being to skip validation, so it doesn't affect all the other policy that cannot have specific elements validated against the signature.

davydsantos commented on validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java:
```diff
@@ -261,7 +272,12 @@ public String getPolicyDescription() {
 		BasicSignatureConstraints basicSignature = getBasicSignatureConstraintsByContext(context);
 		if (basicSignature != null) {
-			return basicSignature.getCryptographic();
+			CryptographicConstraint sigCryptographic = basicSignature.getCryptographic();
```
Is this related to the SignaturePolicyChain?

davydsantos commented on validation-policy/src/main/java/eu/europa/esig/dss/validation/policy/EtsiValidationPolicy.java:
```diff
@@ -270,11 +286,20 @@ public CryptographicConstraint getSignatureCryptographicConstraint(Context conte
 		CertificateConstraints certificateConstraints = getCertificateConstraints(context, subContext);
 		if (certificateConstraints != null) {
-			return certificateConstraints.getCryptographic();
+			CryptographicConstraint certCryptographic = certificateConstraints.getCryptographic();
```
Is this related to the SignaturePolicyChain?

davydsantos reviewed (changes requested):
Some changes seem to be unrelated to this PR...

Commit:
Merge branch 'fraffani/GRBR-463' of https://github.com/davydsantos/dss into fraffani/GRBR-463

fernandoraffani commented:
@davydsantos: I have removed the changes you have noted. 

They were due to differences between master and the version linked to validation-authority that probably need to be merged at some point. However, as they were not part of this fix, I removed them from this PR.

fernandoraffani committed:
Merge branch 'signature-policy' into fraffani/GRBR-463

fernandoraffani commented:
Deprecated: created a new PR: https://github.com/davydsantos/dss/pull/10, that consists of the same changes, but in a branch created from signature-policy branch.

https://jira.corp.docusign.com/browse/GRBR-463
includes https://jira.corp.docusign.com/browse/GRBR-367
